.github/workflows/capgen_unit_tests.yaml

```diff
@@ -0,0 +1,22 @@
+name: Capgen Unit Tests
+
+on:
+  workflow_dispatch:
+  pull_request:
+    types: [opened, synchronize, reopened]
+  push:
+    branches:
+      #Trigger workflow on push to any branch or branch hierarchy:
+      - '**'
+
+jobs:
+  unit_tests:
+    if: github.event_name == 'pull_request' || github.event_name == 'workflow_dispatch' || github.repository == 'NCAR/ccpp-framework'
```

Comment:
I am admittedly weak on the GitHub CI stuff but would this activate on something that happens that is _not_ a PR because of the logical OR statements?
This logic is repeated in other files so I wil just ask this once since it is probably something I just don't understand (yet).

Reply:
The first two statements make sense. Either trigger on a PR *or* when someone runs the workflow manually from the "actions" tab (workflow dispatch). The last statement doesn't make sense to me, because it basically says *always* run when the repo is the NCAR repo. Was the intent to trigger on one of the two preceding statements only if it's the NCAR repo? Then I would have expected something like (pseudo-syntax, because I actually don't know if github actions accept complex logical expressions)
```
if (pull_request || workflow_dispatch) && repo == ncar
```

Reply:
I am still fairly new to GitHub actions so this could possibly be simplified but he intent was to:
1) Run the action on any change to the NCAR repo
2) Run the action on pull requests  (This may need to be updated as the workflows take longer)
3) Allow manually running actions on forked repositories outside of pull requests.
If there's better ways to support this in the workflow files, just let me know!

Reply:
Simply enforce in the branch protection that PRs need to be up to date with develop and you're done.

Reply:
That's a good point about the branch protection rules!  If we implement that along with rules preventing any direct pushes to the NCAR repo (which may already exist?) then I agree that this can all go away.

Reply:
@climbfuji Jesse checked and doesn't have permission to do so, so if you could instead walk me through it, I would appreciate that!